File: week_02/codewars/myeong/200901_WhoLikesIt.cpp
using namespace std;

std::string likes(const std::vector<std::string> &names)
{
  string str=" likes this";
  string strs=" like this";
  string restr;
  int others=names.size()-2;
  
  if(names.size()==0) {
    restr="no one"+str;
  }
  else if(names.size()==1) {
    restr=names[0]+str;
  }  
  else if(names.size()==2) {
    restr=names[0]+" and "+names[1]+strs;
  }
  else if(names.size()==3) {
    restr=names[0]+", "+names[1]+" and "+names[2]+strs;
  }
  else if(names.size()>=4) {
    restr=names[0]+", "+names[1]+" and "+to_string(others)+" others"+strs;
  }
    return restr;
}

File: week_02/codewars/sylee/200901_who_likes_it.py
def likes(names):
    how_many = len(names)
    if how_many >= 4:
        return (
            "{}, {}".format(names[0], names[1])
            + " and "
            + "{} others like this".format(how_many - 2)
        )
    elif how_many == 3:
        return "{}, {} and {} like this".format(names[0], names[1], names[2])
    elif how_many == 2:
        return "{} and {} like this".format(names[0], names[1])
    return "{} likes this".format(names[0]) if how_many  else "no one likes this"